drive/platform-implementations/esp8266_mission_control.cpp

```diff
@@ -55,71 +56,56 @@ class esp8266_mission_control : public mission_control
   {
 
     using namespace std::literals;
-    hal::print(*m_console, "inside imp comm\n");
-    auto http_header_parser = new_http_header_parser();
-    bool write_error = false;
-    bool header_finished = false;
-    bool read_complete = true;
-    auto fill_payload = hal::stream::fill(m_buffer);
-    hal::print(*m_console, "after initial stuff\n");
 
-    if (write_error) {
+    if (m_write_error) {
       hal::print(*m_console, "Reconnecting...\n");
       // Wait 1s before attempting to reconnect
 
       auto result = establish_connection(p_timeout);
       if (!result) {
         return m_commands;
       }
-      write_error = false;
+      m_write_error = false;
     }
-    hal::print(*m_console, "after write error\n");
-
-    if (read_complete) {
+    if (m_read_complete) {
 
       // Send out HTTP GET request
+      
       auto status = m_esp8266->server_write(hal::as_bytes(m_get_request), p_timeout);
-      hal::print(*m_console, "got status\n");
-
+      
       if (!status) {
         hal::print(*m_console, "\nFailed to write to server!\n");
-        write_error = true;
+        hal::print(*m_console, m_get_request);
+        m_write_error = true;
         return m_commands;
       }
 
-      read_complete = false;
-      header_finished = false;
+      m_read_complete = false;
+      m_header_finished = false;
     }
-    hal::print(*m_console, "after read complete\n");
 
     auto received = HAL_CHECK(m_esp8266->server_read(m_buffer)).data;
-    auto remainder = received | http_header_parser.find_header_start |
-                     http_header_parser.find_content_length |
-                     http_header_parser.parse_content_length |
-                     http_header_parser.find_end_of_header;
-    hal::print(*m_console, "after received\n");
-
-    if (!header_finished &&
-        hal::finished(http_header_parser.find_end_of_header)) {
-      auto content_length = http_header_parser.parse_content_length.value();
-      fill_payload = hal::stream::fill(m_buffer, content_length);
-      header_finished = true;
+    auto remainder = received | m_http_header_parser.find_header_start |
+                     m_http_header_parser.find_content_length |
+                     m_http_header_parser.parse_content_length |
+                     m_http_header_parser.find_end_of_header;
+
+    if (!m_header_finished &&
+        hal::finished(m_http_header_parser.find_end_of_header)) {
+      auto content_length = m_http_header_parser.parse_content_length.value();
```

Comment:
```suggestion
      auto& content_length = m_http_header_parser.parse_content_length.value();
```

If you make this change this should start working. auto doesn't copy reference modifiers over. So you have to explicitly make this a reference. 

Value() returns a reference to the internal variable. stream_fill takes a reference/pointer to an integer (honestly bad design but it was meant for handling the single pipeline case without breaking the pipeline). I'll deciding on how to change the API without breaking everyone else's code.